### PR TITLE
feat(ivy): add support for ng-container in the compiler

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

File: packages/compiler/src/render3/view/template.ts [hunk 1]
```diff
@@ -487,13 +488,15 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
 
     const implicit = o.variable(CONTEXT_NAME);
 
-    const createSelfClosingInstruction =
-        !hasStylingInstructions && element.children.length === 0 && element.outputs.length === 0;
+    const createSelfClosingInstruction = !hasStylingInstructions && !isNgContainer &&
```
Comment (JoostK) [inline packages/compiler/src/render3/view/template.ts]: Perhaps add a testcase for `<ng-container></ng-container>` to verify this.

**Review by mhevery — APPROVED** [review]
@alxhub please add your approval as well.

File: packages/compiler/src/render3/view/template.ts (hunk 1 above)
Comment (pkozlowski-opensource) [reply packages/compiler/src/render3/view/template.ts]: Good idea, added a test here: https://github.com/angular/angular/pull/25383/commits/f7166a33cffc7c76c1548aa5356fa66b686b867a#diff-ee14363ac9344a65ec471398a50b0ca2R258

**Review by alxhub — APPROVED** [review]

**Comment by alxhub** [comment]
http://test/OCL:208067786:BASE:208068410:1533835000616:10ded7bc

**Pull request closed without merging**